test/FTSAlgorithmsTest.cpp

```diff
@@ -386,65 +403,189 @@ TEST(FTSAlgorithmsTest, multVarsAggScoresAndTakeTopKContexts) {
-  callFixed(width, wep, nofVars, k, &resW5);
-  ASSERT_EQ(13u, resW5.size());
-  std::sort(std::begin(resW5), std::end(resW5),
+  resW6.clear();
+  callFixed(width, wep, nofVars, k, &resW6);
+  ASSERT_EQ(15u, resW6.size());
+  std::sort(std::begin(resW6), std::end(resW6),
             [](const auto& a, const auto& b) {
-              return a[1] != b[1] ? a[1] > b[1] : a[0] < b[0];
+              return a[0] < b[0] ||
+                     (a[0] == b[0] &&
+                      (a[2] < b[2] ||
+                       (a[2] == b[2] &&
+                        (a[3] < b[3] || (a[3] == b[3] && a[4] < b[4])))));
             });
-  ASSERT_EQ(TextRecordId(0), resW5(0, 0));
-  ASSERT_EQ(IntId(3), resW5(0, 1));
-  ASSERT_EQ(V(0), resW5(0, 2));
-  ASSERT_EQ(V(0), resW5(0, 3));
-  ASSERT_EQ(TextRecordId(1), resW5(1, 0));
-  ASSERT_EQ(IntId(3), resW5(1, 1));
-  ASSERT_EQ(V(0), resW5(1, 2));
-  ASSERT_EQ(V(0), resW5(1, 3));
+  EXPECT_THAT(resW6.getColumn(0),
+              ::testing::ElementsAre(
+                  TextRecordId(0u), TextRecordId(1u), TextRecordId(1u),
+                  TextRecordId(1u), TextRecordId(1u), TextRecordId(1u),
+                  TextRecordId(1u), TextRecordId(2u), TextRecordId(2u),
+                  TextRecordId(2u), TextRecordId(2u), TextRecordId(2u),
+                  TextRecordId(2u), TextRecordId(2u), TextRecordId(2u)));
+  EXPECT_THAT(resW6.getColumn(1),
+              ::testing::ElementsAre(IntId(3u), IntId(3u), IntId(1u), IntId(1u),
+                                     IntId(1u), IntId(1u), IntId(1u), IntId(1u),
+                                     IntId(1u), IntId(1u), IntId(1u), IntId(1u),
+                                     IntId(1u), IntId(1u), IntId(1u)));
+  EXPECT_THAT(resW6.getColumn(2),
+              ::testing::ElementsAre(V(0u), V(0u), V(0u), V(0u), V(1u), V(1u),
+                                     V(1u), V(0u), V(0u), V(0u), V(2u), V(2u),
+                                     V(2u), V(2u), V(2u)));
+  EXPECT_THAT(resW6.getColumn(3),
+              ::testing::ElementsAre(V(0u), V(0u), V(1u), V(1u), V(0u), V(0u),
+                                     V(1u), V(2u), V(2u), V(2u), V(0u), V(0u),
+                                     V(0u), V(2u), V(2u)));
+  EXPECT_THAT(
+      resW6.getColumn(4),
+      ::testing::ElementsAre(
+          WordVocabId(1u), WordVocabId(1u), WordVocabId(1u), WordVocabId(2u),
+          WordVocabId(1u), WordVocabId(2u), WordVocabId(2u), WordVocabId(1u),
+          WordVocabId(3u), WordVocabId(5u), WordVocabId(1u), WordVocabId(3u),
+          WordVocabId(5u), WordVocabId(3u), WordVocabId(5u)));
+  EXPECT_THAT(
+      resW6.getColumn(5),
+      ::testing::ElementsAre(
+          WordVocabId(6u), WordVocabId(9u), WordVocabId(9u), WordVocabId(8u),
+          WordVocabId(9u), WordVocabId(8u), WordVocabId(8u), WordVocabId(8u),
+          WordVocabId(7u), WordVocabId(9u), WordVocabId(8u), WordVocabId(7u),
+          WordVocabId(9u), WordVocabId(7u), WordVocabId(9u)));
```

Comment:
You can also create shorter aliases/creation functions for
`WordVocabId` and `TextRecordId` in that file, then it becomes much more readable.

Reply:
(I think `T` for `TextRecordId` even already exists.